landoapi/hgexports.py

```diff
@@ -274,85 +287,83 @@ def __init__(self, fileobj: io.BytesIO):
         super().__init__(fileobj)
         self.parse_patch()
 
-    def parse_patch(self):
-        """Parse the Git patch file for headers and diff content."""
-        mail_from_line = next(self.patch)
-        if not mail_from_line.startswith(b"From "):
-            raise ValueError("Patch is malformed, first line is not `From `.")
-
-        author_from_line = next(self.patch)
-        if not author_from_line.startswith(b"From: "):
-            raise ValueError("Patch is malformed, second line is not `From:`.")
-        self.set_header(
-            "From", author_from_line.removeprefix(b"From: ").removesuffix(b"\n")
-        )
+    def get_header(self, name: bytes | str) -> Optional[str]:
+        """Get the headers from the message."""
+        name = name.decode("utf-8") if isinstance(name, bytes) else name
 
-        date_line = next(self.patch)
-        if not date_line.startswith(b"Date: "):
-            raise ValueError("Patch is malformed, third line is not `Date:`.")
-        self.set_header("Date", date_line.removeprefix(b"Date: ").removesuffix(b"\n"))
-
-        # Get the main `Subject` header line.
-        subject_line = next(self.patch)
-        if not subject_line.startswith(b"Subject: "):
-            raise ValueError("Patch is malformed, fourth line is not `Subject:`.")
-
-        # Clean up the first subject line.
-        subject_line = (
-            subject_line.removeprefix(b"Subject: ")
-            .removeprefix(b"[PATCH] ")
-            .removesuffix(b"\n")
-        )
+        # `EmailMessage` will return `None` if the header isn't found.
+        return self.message[name]
+
+    def verify_content(self, content: str) -> str:
```

Comment:
It would be good if this was renamed -- since it is doing a more than verifying (i.e., setting `self.commit_message` and returning `diff_lines`.) Maybe something like `verify_and_set_diff_content` or something with a verb implying things are being modified.

Same comment applies to the docstring.